deployer/scripts/functions.sh

```diff
@@ -30,11 +30,23 @@ function setup_certificate {
 function handle_previous_deployment() {
   if [ "$mode" = "refresh" ]; then
     echo "Deleting any previous deployment (leaving route and PVCs)"
-    oc delete rc,svc,pod,sa,templates,secrets --selector="metrics-infra"
+    oc delete rc,svc,pod,sa,templates,secrets --selector="metrics-infra" --ignore-not-found=true
   elif [ "$redeploy" = true ] || [ "$mode" = remove ]; then
     echo "Deleting any previous deployment"
-    oc delete --grace-period=0 all,sa,templates,secrets --selector="metrics-infra"
-    oc delete pvc --selector="metrics-infra"
+    # We don't want to delete ourselves, but we do want to remove old deployers
+    # Remove our label so that we are not immediately deleted.
+    echo "POD_NAME ${POD_NAME:-}"
+    if [ -n "${POD_NAME:-}" ]; then
+      oc label pod ${POD_NAME} metrics-infra-
+    fi
```

Comment:
Personally I would shorten to:

    [ -n "${POD_NAME:-}" ] && oc label pod ${POD_NAME} metrics-infra-